lldb/source/Interpreter/CommandObject.cpp

```diff
@@ -754,23 +754,29 @@ const char *CommandObject::GetArgumentDescriptionAsCString(
   return g_argument_table[arg_type].help_text;
 }
 
-Target &CommandObject::GetDummyTarget() {
+Target &CommandObject::GetTarget(bool dummy) {
+  // Always return the dummy target if explicitly requested.
+  if (dummy)
+    return m_interpreter.GetDebugger().GetDummyTarget();
+
+  // Prefer the frozen execution context in the command object.
+  if (Target *target = m_exe_ctx.GetTargetPtr())
+    return *target;
+
+  // Fallback to the command interpreter's execution context in case we get
+  // called after DoExecute has finished. For example, when doing multi-line
+  // expression that uses an input reader or breakpoint callbacks.
+  if (Target *target = m_interpreter.GetExecutionContext().GetTargetPtr())
+    return *target;
+
+  // Finally, if we have no other target, get the selected target.
+  if (TargetSP target_sp = m_interpreter.GetDebugger().GetSelectedTarget())
+    return *target_sp;
+
+  // We only have the dummy target.
```

Comment:
I don't think it's worth using an optional here, but passing out a reference here means that if you called this with dummy == false, and there was only the dummy target, we would still return the Dummy target.

That seems unexpected - but maybe just worth a comment?

Reply:
The `dummy` argument is meant to indicate that we always want the dummy target. Would it be clearer if I called it `force_dummy` or `dummy_only` or something?

Reply:
Or `prefer_dummy` like the original signature? I shortened it to facilitate the inline comments, but I can change it back.

Reply:
I kind of like that better, you shouldn't be dialing up the dummy target directly unless you really mean it, so making that stand out a bit more in the code might be good.

Reply:
Alright, I removed the argument from `GetTarget` and added `GetDummyTarget` again.

Reply:
Yeah, that makes the case where people are dialing it up from an option slightly more verbose, but much clearer what's going on.